File: GeneratedEntityReaderClasses/Components/SineWaveComponent.cs
namespace GeneratedNoitaClasses;
public class SineWaveComponent : NoitaComponentBase,  iNoitaType<SineWaveComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat sinewave_freq { get; set; } = new();
	public NoitaFloat sinewave_m { get; set; } = new();
	public NoitaInt lifetime { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		sinewave_freq.Read(s);
		sinewave_m.Read(s);
		lifetime.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		sinewave_freq.Write(s);
		sinewave_m.Write(s);
		lifetime.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/SineWaveComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class SineWaveComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    sinewave_freq:NoitaFloat = field(default_factory=NoitaFloat)
    sinewave_m:NoitaFloat = field(default_factory=NoitaFloat)
    lifetime:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.sinewave_freq.Read(s)
        self.sinewave_m.Read(s)
        self.lifetime.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.sinewave_freq.Write(s)
        self.sinewave_m.Write(s)
        self.lifetime.Write(s)

